cli/pkg/helm/client.go

```diff
@@ -0,0 +1,557 @@
+	return c.getChartValues(name)
+}
+
+// Get chart metadata
+func (c *HelmClient) InspectChartYaml(name string) ([]byte, error) {
+	return c.getChartYaml(name)
+}
+
+// UninstallRelease uninstalls the provided release
+func (c *HelmClient) UninstallRelease(spec *ChartSpec) error {
+	return c.uninstallRelease(spec)
+}
+
+// install lints and installs the provided chart
+func (c *HelmClient) install(spec *ChartSpec) (*release.Release, error) {
+	release := &release.Release{}
+	client := action.NewInstall(c.ActionConfig)
+	mergeInstallOptions(spec, client)
+
+	if client.Version == "" {
+		client.Version = ">0.0.0-0"
+	}
+
+	helmChart, chartPath, err := c.getChart(spec.ChartName, &client.ChartPathOptions)
+	if err != nil {
+		return release, err
+	}
+
+	if helmChart.Metadata.Type != "" && helmChart.Metadata.Type != "application" {
+		return release, fmt.Errorf(
+			"chart %q has an unsupported type and is not installable: %q",
+			helmChart.Metadata.Name,
+			helmChart.Metadata.Type,
+		)
+	}
+
+	d := &updateDeps{
+		helmChart:        helmChart,
+		spec:             spec,
+		chartPath:        &chartPath,
+		chartPathOptions: &client.ChartPathOptions,
+	}
+	err = c.updateDependencies(d)
+	if err != nil {
+		return release, fmt.Errorf("failed to update helm chart dependencies %v", err)
+	}
+
+	values, err := spec.GetValuesMap()
+	if err != nil {
+		return release, err
+	}
+
+	if c.linting {
+		err = c.lint(chartPath, values)
+		if err != nil {
+			return release, err
+		}
+	}
+
+	client.Namespace = c.Settings.Namespace()
+	// create namespace if it doesn't exist
+	client.CreateNamespace = true
+	release, err = client.Run(helmChart, values)
+	if err != nil {
+		return release, err
+	}
+
+	log.Printf("release installed successfully: %s/%s-%s", release.Name, release.Name, release.Chart.Metadata.Version)
+
+	return release, nil
+}
+
+type updateDeps struct {
+	helmChart        *chart.Chart
+	spec             *ChartSpec
+	chartPath        *string
+	chartPathOptions *action.ChartPathOptions
+}
+
+func (c *HelmClient) updateDependencies(details *updateDeps) error {
+	if req := details.helmChart.Metadata.Dependencies; req != nil {
+		if err := action.CheckDependencies(details.helmChart, req); err != nil {
+			if details.spec.DependencyUpdate {
+				man := &downloader.Manager{
+					ChartPath:        *details.chartPath,
+					Keyring:          details.chartPathOptions.Keyring,
+					SkipUpdate:       false,
+					Getters:          c.Providers,
+					RepositoryConfig: c.Settings.RepositoryConfig,
+					RepositoryCache:  c.Settings.RepositoryCache,
+					Out:              os.Stdout,
+				}
+				if err := man.Update(); err != nil {
+					return err
+				}
+
+				// as chart dependencies are updated fetch the chart again
+				details.helmChart, *details.chartPath, err = c.getChart(details.spec.ChartName, details.chartPathOptions)
+				if err != nil {
+					return err
+				}
+			} else {
+				return err
+			}
```

Comment:
Remove else by handling errors first.
```suggestion
			if !details.spec.DependencyUpdate {
				return err
			}
```